test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
+        name: 'Test1',
+      };
+
+      saveMock.mockResolvedValue(roles[0]);
+
+      const result = await roleService.createRole(input);
 
-    rolesRepository
-      .update('ae032b1b-cc3c-4e44-9197-276ca877a7f8', input)
-      .returns(Promise.resolve(Arg.any()));
+      expect(saveMock).toBeCalledWith(input);
 
-    const resp = await roleService.updateRole(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    expect(resp).toEqual(roles[0]);
+      expect(result).toEqual(roles[0]);
+    });
   });
 
-  it('should add permissions to a role', async () => {
-    const request = [
-      {
-        roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-        permissionId: '2b33268a-7ff5-4cac-a87a-6bfc4430d34c',
-      },
-    ];
-    const input: UpdateRolePermissionInput = {
-      permissions: ['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'],
-    };
-
-    permissionRepository
-      .findByIds(['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'])
-      .resolves(permissions);
-
-    rolePermissionRepository.create(request).returns(request);
-
-    permissionRepository
-      .createQueryBuilder('permission')
-      .returns(permissionQueryBuilder);
-    permissionQueryBuilder
-      .leftJoinAndSelect(
-        RolePermission,
-        'rolePermission',
-        'permission.id = rolePermission.permissionId',
-      )
-      .returns(permissionQueryBuilder);
-
-    permissionQueryBuilder
-      .where('rolePermission.roleId = :roleId', {
-        roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      })
-      .returns(permissionQueryBuilder);
-
-    permissionQueryBuilder.getMany().resolves(permissions);
-
-    connectionMock.transaction(Arg.any()).resolves(request);
-
-    const resp = await roleService.updateRolePermissions(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    expect(resp).toEqual(permissions);
+  describe('updateRole', () => {
+    it('should update a role for a valid role id', async () => {
+      const input: UpdateRoleInput = {
+        name: 'Test1',
+      };
+
+      updateMock.mockResolvedValue(updateResult);
+
+      const result = await roleService.updateRole(
+        'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        input,
+      );
+
+      expect(updateMock).toBeCalledWith(VALID_ROLE_ID, input);
+
+      expect(result).toEqual(true);
+    });
   });
 
-  it('should throw exception when adding invalid permissions to a role', async () => {
-    const input: UpdateRolePermissionInput = {
-      permissions: ['3e9e78c9-3fcd-4eed-b027-62f794680b03'],
-    };
-
-    permissionRepository
-      .findByIds(['3e9e78c9-3fcd-4eed-b027-62f794680b03'])
-      .resolves([]);
-
-    const resp = roleService.updateRolePermissions(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    await expect(resp).rejects.toThrowError(
-      new PermissionNotFoundException(
-        ['3e9e78c9-3fcd-4eed-b027-62f794680b03'].toString(),
-      ),
-    );
+  describe('deleteRole', () => {
```

Comment:
Test fails.

Reply:
Will fix service related tests in a separate PR.